olp-cpp-sdk-core/src/client/HRN.cpp

```diff
@@ -19,84 +19,101 @@
 
 #include "olp/core/client/HRN.h"
 
-#include <cassert>
 #include <sstream>
 
+#include <olp/core/logging/Log.h>
+#include <olp/core/porting/make_unique.h>
 #include "Tokenizer.h"
 
+namespace {
+constexpr auto kLogTag = "HRN";
+static const std::string kDataTag = "data";
+static const std::string kPipelineTag = "pipeline";
+static const std::string kSchemaTag = "schema";
+static const std::string kHRNTag = "hrn:";
+static constexpr char kSeparator = ':';
+}  // namespace
+
 namespace olp {
 namespace client {
 
 std::string HRN::ToString() const {
   std::ostringstream ret;
-  std::string generic_part = ":" + region + ":" + account + ":";
-  ret << "hrn:" << partition << ":";
+  auto generic_part = kSeparator + region + kSeparator + account + kSeparator;
```

Comment:
```suggestion
  const auto generic_part = kSeparator + region + kSeparator + account + kSeparator;
```
Prefer declaring the variable as const by default, unless there is a reason to change it (Con1: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rconst-immutable)